srv/salt/_modules/cephprocesses.py

```diff
@@ -26,13 +26,24 @@ def check():
 
     ret = True
     if 'roles' in __pillar__:
```

Comment:
There is too much nesting here, which  can make code brittle. I would suggest using a Fail-Fast technique here to reduce this.

There is also a fair bit of code reuse between the if and else blocks with the exception of the print statements. These statements don't indicate that they are pillar-only, so why not print them out for both cases or never at all?
 Additionally, using `result` or `ret` (or their variants) as a variable names is very poor process.

Without checking too much for correctness of my snippet ...
```py
can_continue = True
if 'roles' not in __pillar__:
    return True # can continue
roles = __pillar__['roles']
if only_for:
    roles = only_for['roles']
for role in roles:
    log.info('...')
    for process in processes[role]:
        pid = __salt__['status.pid'](process)
        print "ON ROLE: {}".format(role)
        print "RESULT: {}".format(pid)
        if not pid:
            log.error('...')
            can_continue = False
return can_continue
```

`if not pid` checks if `pid` is empty, and it works, but I think a better test altogether can be constructed. This relies on salt to always output a null string with no extra spaces or newlines for the rest of time, and it relies on `status.pid` to always return a pid or nothing for the rest of time. To protect ourselves from any possible modifications to salt in the future, we are better of testing whether the result we get is one and only one integer value and returning error otherwise.

```py
if not pid.isdigit():
    log.error('...')
```